### PR TITLE
feat: added lines of code that makes a backup of nix before it is used so user can reset nix easily

### PR DESCRIPTION
**Pull request by dnkmmr69420**
title says everything

### CODE REVIEW

File: usr/bin/ublue-nix-install [hunk 1]
```diff
@@ -95,6 +95,29 @@ echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.con
...
+sudo rm -rf /nix/*
+sudo mkdir -p /nix
+sudo cp -R /opt/nb/nix/* /nix/
+sudo restorecon -RF /nix
+sudo echo "Nix has been resetted. Reboot for changes to apply."
+EOF
+
+sudo chmod a+x /opt/nb/reset-nix
+
+sudo echo "Finnished installing nix backup"
```
Comment (EinoHR) [inline usr/bin/ublue-nix-install:117]: Finnished -> Finished

File: usr/bin/ublue-nix-install [hunk 2]
```diff
@@ -95,6 +95,29 @@ echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.con
...
+sudo cp -R /nix /opt/nb
+
+sudo tee /opt/nb/reset-nix <<EOF
+#!/bin/bash
+sudo echo "Resetting nix..."
+sudo rm -rf /nix/*
+sudo mkdir -p /nix
+sudo cp -R /opt/nb/nix/* /nix/
+sudo restorecon -RF /nix
+sudo echo "Nix has been resetted. Reboot for changes to apply."
```
Comment (bketelsen) [inline usr/bin/ublue-nix-install:112]: resetted -> reset

**Review by bketelsen — APPROVED** [review]
LGTM

File: usr/bin/ublue-nix-install [hunk 3]
```diff
@@ -95,6 +95,29 @@ echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.con
 
 sleep 1
 
+# nb stands for nix backup
+
+echo "Installing nix backup"
+
+sudo mkdir /opt/nb
```
Comment (bketelsen) [inline usr/bin/ublue-nix-install:102]: I'd rather this had a more meaningful name like `/opt/nixbackup`

Comment (dnkmmr69420) [reply usr/bin/ublue-nix-install:102]: yeah you could change it to that but I did nb since it is very short

Comment (castrojo) [inline usr/bin/ublue-nix-install:102]:
```suggestion
sudo mkdir /opt/nixbackup
```

File: usr/bin/ublue-nix-install [hunk 4]
```diff
@@ -95,6 +95,29 @@ echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.con
...
+sudo cp -R /nix /opt/nb
+
+sudo tee /opt/nb/reset-nix <<EOF
+#!/bin/bash
+sudo echo "Resetting nix..."
+sudo rm -rf /nix/*
+sudo mkdir -p /nix
+sudo cp -R /opt/nb/nix/* /nix/
```
Comment (castrojo) [inline usr/bin/ublue-nix-install:110]:
```suggestion
sudo cp -R /opt/nixbackup/nix/* /nix/
```

File: usr/bin/ublue-nix-install [hunk 5]
```diff
@@ -95,6 +95,29 @@ echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.con
...
+sudo restorecon -RF /nix
+sudo echo "Nix has been reset. Reboot for changes to apply."
+EOF
+
+sudo chmod a+x /opt/nb/reset-nix
```
Comment (castrojo) [inline usr/bin/ublue-nix-install:115]:
```suggestion
sudo chmod a+x /opt/nixbackup/reset-nix
```

**Review by castrojo — CHANGES_REQUESTED** [review]
I want to merge this but I think we should be explicit here just in case (thinking of new users). I've proposed the changes, LGTM, thanks!

**Review by castrojo — APPROVED** [review]

**Comment by castrojo** [comment]
Yeah this won't merge unless it passes the semantic check, needs to be done in git and not just the title in github.

**Pull request merged**